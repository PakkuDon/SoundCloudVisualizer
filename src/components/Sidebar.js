import React from 'react'

const Sidebar = (props) => (
  <React.Fragment>
    <input id="toggle-sidebar" type="checkbox" />
    <div id="sidebar">
      {props.children}
    </div>
    <label for="toggle-sidebar">&#9664;</label>
  </React.Fragment>
)

export default Sidebar
