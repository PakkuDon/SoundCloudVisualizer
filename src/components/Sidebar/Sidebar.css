@import "../../styles/colors";

.root {
  background-color: $blue200;
  display: flex;
  flex-direction: row;
  height: 100%;
  max-width: 25%;
}

.sidebarWrapper {
  color: $white;
  padding: 5px;
}

.toggleSidebar {
  background-color: inherit;
  border-color: transparent;
  color: $white;
  padding: 2px;
  border-radius: 0;
  align-self: center;
  height: 100%;
}
