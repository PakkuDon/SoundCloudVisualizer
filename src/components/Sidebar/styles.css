.root {
  display: flex;
  flex-direction: row;
  height: 100%;
}

.sidebarWrapper {
  flex: 1 0 auto;
  background-color: rgba(0, 160, 230, 0.1);
  color: #FFF;
  padding: 5px;
}

.toggleSidebar {
  background-color: rgba(0, 160, 230, 0.25);
  border-color: transparent;
  color: #FFF;
  padding: 2px;
  border-radius: 0;
  align-self: center;
  height: 100%;
}
