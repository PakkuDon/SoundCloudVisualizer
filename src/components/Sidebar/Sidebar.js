import React, { useState } from "react"
import styles from "./Sidebar.css"

const Sidebar = props => {
  const [isOpen, toggleSidebar] = useState(true)

  return (
    <div className={styles.root} data-testid="sidebar">
      {isOpen && <div className={styles.sidebarWrapper}>{props.children}</div>}
      <button
        className={styles.toggleSidebar}
        title="Toggle sidebar"
        onClick={() => toggleSidebar(!isOpen)}
      >
        &#9664;
      </button>
    </div>
  )
}

export default Sidebar
