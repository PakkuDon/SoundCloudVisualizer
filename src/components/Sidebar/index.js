import React, { useState } from 'react'
import styles from './styles.css'

const Sidebar = (props) => {
  const [isOpen, toggleSidebar] = useState(true)

  return (
    <React.Fragment>
      {isOpen && (
        <div className={styles.root}>
          {props.children}
        </div>
      )}
      <button
        className={styles.toggleSidebar}
        onClick={() => toggleSidebar(!isOpen)}
      >
        &#9664;
      </button>
    </React.Fragment>
  )
}

export default Sidebar
