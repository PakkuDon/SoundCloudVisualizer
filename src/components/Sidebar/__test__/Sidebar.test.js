import React from "react"
import { shallow } from "enzyme"

import Sidebar from "../Sidebar"

describe("Sidebar", () => {
  it("renders children", () => {
    const children = <div>Hello world</div>
    const wrapper = shallow(<Sidebar>{children}</Sidebar>)

    expect(wrapper.contains(children)).toBe(true)
  })

  describe("when collapsed", () => {
    it("hides children", () => {
      const children = <div>Hello world</div>
      const wrapper = shallow(<Sidebar>{children}</Sidebar>)

      wrapper.find('button').simulate('click')

      expect(wrapper.contains(children)).toBe(false)
    })
  })
})
