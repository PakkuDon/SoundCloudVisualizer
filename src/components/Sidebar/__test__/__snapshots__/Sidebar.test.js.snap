// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders children 1`] = `
<div
  className="root"
  data-testid="sidebar"
>
  <div
    className="sidebarWrapper"
  >
    <div>
      Hello world
    </div>
  </div>
  <button
    className="toggleSidebar"
    onClick={[Function]}
  >
    â—€
  </button>
</div>
`;
